src/main/java/org/gnode/nix/Block.java

```diff
@@ -20,20 +24,20 @@
 
     /**
      * Constructor that creates an uninitialized Block.
-     * <p/>
+     * <p>
      * Calling any method on an uninitialized block will throw a {@link java.lang.RuntimeException}.
      */
     public Block() {
         allocate();
     }
 
-    private native void allocate();
+    private synchronized native void allocate();
```

Comment:
Synchronize modifier can be removed, its presence suggests that the methods are thread-safe which is not the case.